de.fhdo.lemma.servicedsl.extractor/src/de/fhdo/lemma/servicedsl/extractor/ServiceDslExtractor.xtend

```diff
@@ -0,0 +1,272 @@
+            «ENDFOR»
+        }'''
+    }
+
+    /**
+     * Extract Endpoint
+     */
+    private def generate(Endpoint endpoint) {
+    	// Formatting is kinda ugly because otherwise xtend's string 
+    	// templates have additional not intended linebreaks
+        return '''«FOR ep: endpoint.protocols SEPARATOR '; '»
+        	«ep.importedProtocol.generate»:«
+        ENDFOR
+        »«FOR ea: endpoint.addresses SEPARATOR ', '
+        	»"«ea»"«
+        ENDFOR»;'''
+    }
+
+    /**
+     * Extract Operation
+     */
+    private def generate(Operation operation) {
+        val endpoints ='''
+        @endpoints(«FOR e: operation.endpoints»«e.generate»«ENDFOR»)
+        '''
+        
+        val aspects = '''
+        «FOR  a: operation.aspects»«a.generate»«ENDFOR»
+        '''
+        
+        val parameters = String.join(", ", operation.parameters.map[generate])
+
+        val comment = if (operation.apiOperationComment !== null) {
+            '''
+            ---
+            «operation.apiOperationComment.comment»
+            «FOR param : operation.parameters.filter[!it.isOptional]»
+            @required «param.name» [INSERT PARAMETER DESC HERE]
+            «ENDFOR»
+            «FOR param : operation.parameters.filter[it.isOptional]»
+            @param «param.name» [INSERT PARAMETER DESC HERE]
+            «ENDFOR»            
+            ---
+            '''
+        }
+        
+        return '''«comment»«endpoints»«aspects»«operation.name»(«parameters»);'''
+    }
+
+    /**
+     * Extract Parameter
+     */
+    private def generate(Parameter parameter) {
+    	// Formatting is kinda ugly because otherwise xtend's string 
+    	// templates have additional not intended linebreaks
+        return '''«
+        FOR a : parameter.aspects SEPARATOR ' '
+        	»«a.generate»«
+        ENDFOR
+        » «parameter.communicationType.generate» «parameter.exchangePattern.generate» «
+        parameter.name» : «parameter.generateType
+        »'''
+    }
+    
+    /**
+     * Extract CommunicationType
+     */
+    private def generate(CommunicationType type) {
+        return switch(type) {
+            case ASYNCHRONOUS: "async"
+            case SYNCHRONOUS: "sync"
+            default: null
+        }
+    }
+
+    /**
+     * Extract ExchangePattern
+     */
+    private def generate(ExchangePattern pattern) {
+        return switch(pattern) {
+            case IN: "in"
+            case OUT: "out"
+            case INOUT: "inout"
+            default: null
+        }
+    }
+
+    /**
+     * Extract ImportedServiceAspect
+     */
+    private def generate(ImportedServiceAspect aspect) {
+        return '''@«aspect.importedAspect.technology.name»::«
+        FOR s : aspect.importedAspect.getQualifiedNameParts(false, true) SEPARATOR '.'»«s
+        »«ENDFOR»'''
+    }
+
+    /**
+     * Extract Parameter
+     */
+    private def generateType(Parameter parameter) {
+        return if (parameter.primitiveType !== null)
```

Comment:
The `Parameter` metamodel concept has the `effectiveType` operation, which you can use here as it does the same as the `if`-statement but in a slightly safer manner (cf. https://github.com/SeelabFhdo/lemma/blob/ebd5cc295c6993068eb1fe7c7b97c9790769359c/de.fhdo.lemma.servicedsl.metamodel/model/ServiceViewpointModel.xcore#L1275).